legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-postgres/src/main/java/org/finos/legend/engine/persistence/components/relational/postgres/PostgresSink.java

```diff
@@ -0,0 +1,166 @@
+        explicitDataTypeMapping.put(DataType.TINYINT, new HashSet<>(Arrays.asList(DataType.SMALLINT, DataType.INTEGER, DataType.INT, DataType.BIGINT, DataType.FLOAT, DataType.DOUBLE, DataType.DECIMAL, DataType.NUMERIC)));
+        explicitDataTypeMapping.put(DataType.SMALLINT, new HashSet<>(Arrays.asList(DataType.INTEGER, DataType.INT, DataType.BIGINT, DataType.FLOAT, DataType.DOUBLE, DataType.DECIMAL, DataType.NUMERIC)));
+        explicitDataTypeMapping.put(DataType.INTEGER, new HashSet<>(Arrays.asList(DataType.BIGINT, DataType.FLOAT, DataType.DOUBLE, DataType.DECIMAL, DataType.NUMERIC)));
+        explicitDataTypeMapping.put(DataType.BIGINT, new HashSet<>(Arrays.asList(DataType.DECIMAL, DataType.NUMERIC)));
+        explicitDataTypeMapping.put(DataType.REAL, new HashSet<>(Arrays.asList(DataType.DOUBLE, DataType.DECIMAL, DataType.NUMERIC)));
+        explicitDataTypeMapping.put(DataType.DOUBLE, new HashSet<>(Arrays.asList(DataType.DECIMAL, DataType.NUMERIC)));
+        explicitDataTypeMapping.put(DataType.CHAR, new HashSet<>(Arrays.asList(DataType.VARCHAR, DataType.LONGTEXT, DataType.STRING)));
+        explicitDataTypeMapping.put(DataType.VARCHAR, Collections.singleton(DataType.LONGTEXT));
+        EXPLICIT_DATA_TYPE_MAPPING = Collections.unmodifiableMap(explicitDataTypeMapping);
+
+        INSTANCE = new PostgresSink();
+    }
+
+    public static RelationalSink get()
+    {
+        return INSTANCE;
+    }
+
+    public static Connection createConnection(String user, String pwd, String jdbcUrl)
+    {
+        try
+        {
+            return DriverManager.getConnection(jdbcUrl, user, pwd);
+        }
+        catch (SQLException e)
+        {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    private PostgresSink()
+    {
+        super(
+            CAPABILITIES,
+            IMPLICIT_DATA_TYPE_MAPPING,
+            EXPLICIT_DATA_TYPE_MAPPING,
+            SqlGenUtils.QUOTE_IDENTIFIER,
+            LOGICAL_PLAN_VISITOR_BY_CLASS,
+            (executor, sink, dataset) -> sink.doesTableExist(dataset),
+            (executor, sink, dataset) -> sink.validateDatasetSchema(dataset, new PostgresDataTypeMapping()),
+            (executor, sink, dataset) -> sink.constructDatasetFromDatabase(dataset, new PostgresJdbcPropertiesToLogicalDataTypeMapping()));
+    }
+
+    @Override
+    public Executor<SqlGen, TabularData, SqlPlan> getRelationalExecutor(RelationalConnection relationalConnection)
+    {
+        if (relationalConnection instanceof JdbcConnection)
+        {
+            JdbcConnection jdbcConnection = (JdbcConnection) relationalConnection;
+            return new RelationalExecutor(this, JdbcHelper.of(jdbcConnection.connection()));
+        }
+        else
+        {
+            throw new UnsupportedOperationException("Only JdbcConnection is supported for Postgres Sink");
+        }
+    }
+
+    @Override
+    public Optional<Optimizer> optimizerForCaseConversion(CaseConversion caseConversion)
```

Comment:
Possible to move this to AnsiSink for all the sinks

Reply:
Sure, changed! Removed the overriding method from Postgres, H2, Bigquery and Memsql; retained the one in Snowflake as it is using a different optimizer